### PR TITLE
Remove unnecessary new lines after error message.

### LINKED ISSUE #260: Error output has trailing new lines
There is a lot of whitespace after the last error message.
```
ᐅ build/host/sdk/bin/toit.run tests/negative/as2_test.toit 
As check failed: a B is not a A.
  0: main                      tests/negative/as2_test.toit:8:5
  1: __entry__.<lambda>        <sdk>/core/entry.toit:48:20


ᐅ
```

### PR DESCRIPTION
**Pull request by floitsch**
Fixes #260.

### CODE REVIEW

File: lib/core/print.toit [hunk 1]
```diff
@@ -46,12 +46,30 @@ print_ object:
...
+  write_string_on_stdout_ message
+  write_string_on_stdout_ "\n"
+
+// Dumps the string $message on stdout without a newline and flushes.
+write_on_stdout_ message:
+  write_string_on_stdout_ message.stringify
+
+// Dumps the string $message on stdout without a newline and flushes.
+write_string_on_stdout_ message:
+  #primitive.core.write_string_on_stdout
```
Comment (kasperl) [inline lib/core/print.toit:58]: I'd consider passing a parameter to the write-primitives instead to only get one flushing. One print -> one write.

Comment (floitsch) [reply lib/core/print.toit:79]: done.

File: lib/core/print.toit [hunk 2]
```diff
@@ -36,22 +36,78 @@ Does not yield the currently running task.
...
-  #primitive.core.print_string_on_stdout
+  write_string_on_stdout_ message true
+
+/**
+Dumps the string of $object on stdout and flushes it.
+If $add_newline is true adds a "\n" to the output.
+
+Does not yield the currently running task.
+*/
+write_on_stdout_ object add_newline :
```
Comment (kasperl) [inline lib/core/print.toit:69]: Remove space before :.

Comment (kasperl) [reply lib/core/print.toit:69]: Is it worth having this extra indirection instead of just letting the two callers call stringify?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/core/print.toit (hunk 2 above)
Comment (floitsch) [reply lib/core/print.toit]: Removed the space.

Will do a follow-up PR to remove the indirections.

**Pull request merged**